lib/Macros/Sources/SwiftMacros/TaskLocalMacro.swift

```diff
@@ -0,0 +1,204 @@
+///
+/// It introduces a peer `static let $name: TaskLocal<Type>` as well as a getter
+/// that assesses accesses the task local storage.
+public enum TaskLocalMacro {}
+
+extension TaskLocalMacro: PeerMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    providingPeersOf declaration: some DeclSyntaxProtocol,
+    in context: some MacroExpansionContext
+  ) throws -> [DeclSyntax] {
+    guard let varDecl = try requireVar(declaration, diagnose: false) else {
+      return []
+    }
+    guard try requireStaticContext(varDecl, in: context, diagnose: false) else {
+      return []
+    }
+
+    guard let firstBinding = varDecl.bindings.first else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have declared binding", id: .incompatibleDecl)
+    }
+
+    guard let name = firstBinding.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have name", id: .incompatibleDecl)
+    }
+
+    let type = firstBinding.typeAnnotation?.type
+    let explicitType: String
+    if let type {
+      explicitType = ": TaskLocal<\(type.trimmed)>"
+    } else {
+      explicitType = ""
+    }
+
+    let initialValue: Any
+    if let initializerValue = firstBinding.initializer?.value {
+      initialValue = initializerValue
+    } else if let type, type.isOptional {
+      initialValue = "nil"
+    } else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have default value, or be optional", id: .mustBeVar)
+    }
+
+    // If the property is global, do not prefix the synthesised decl with 'static'
+    let isGlobal = context.lexicalContext.isEmpty
+    let staticKeyword: String
+    if isGlobal {
+      staticKeyword = ""
+    } else {
+      staticKeyword = "static "
+    }
+
+    return [
+      """
+      \(raw: staticKeyword)let $\(name)\(raw: explicitType) = TaskLocal(wrappedValue: \(raw: initialValue))
+      """
+    ]
+  }
+}
+
+extension TaskLocalMacro: AccessorMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    providingAccessorsOf declaration: some DeclSyntaxProtocol,
+    in context: some MacroExpansionContext
+  ) throws -> [AccessorDeclSyntax] {
+    // We very specifically have to fail and diagnose in the accessor macro,
+    // rather than in the peer macro, since returning [] from the accessor
+    // macro adds another series of errors about it missing to emit a decl.
+    guard let varDecl = try requireVar(declaration) else {
+      return []
+    }
+    try requireStaticContext(varDecl, in: context)
+
+    guard let firstBinding = varDecl.bindings.first else {
+      return [] // TODO: make error
+    }
+
+    guard let name = firstBinding.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+      return [] // TODO: make error
+    }
```

Comment:
Do you want to address these TODOs in follow-up PRs?

Reply:
Thank you, yeah actually let me just remove them -- we have a situation here where the same macro has two roles and one of them already is emitting those errors, so those should be silent here